contrib/btrfs_print_fs_uuids_cache.py

```diff
@@ -0,0 +1,201 @@
+    - This dumps all the structures. The output will be enormous,
+    so it's recommended to redirect it to a file and read it from there.
+"""
+
+import drgn
+import argparse
+
+from drgn.helpers.common import *
+from drgn.helpers.linux import (
+    list_for_each_entry,
+    for_each_possible_cpu,
+    per_cpu_ptr,
+)
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-d", "--dump", required = False, help = "Dump all structure data")
+
+args = parser.parse_args()
+
+def format(uuid_list):
+    """Helper to format uuid from list representation"""
+    part = ""
+    for ele in uuid_list:
+        part += "%02x" %ele
+    return part
+
+def get_uuid_from_list(uuid_list):
+    """Helper to format uuid from list representation"""
+    uuid = ""
+    uuid += format(uuid_list[0:4]) + "-"
+    uuid += format(uuid_list[4:6]) + "-"
+    uuid += format(uuid_list[6:8]) + "-"
+    uuid += format(uuid_list[8:10]) + "-"
+    uuid += format(uuid_list[10:])
+    return uuid
+
+def print_mount_info(mnt):
+    """Print key fields of `struct mount`"""
+    print(f"Mount Info:")
+    print(f"\tMount Point: {mnt.mnt_mountpoint.d_iname.string_().decode('utf-8')}")
+    print(f"\tVFS Mount Flags: {mnt.mnt.mnt_flags}")
+    print(f"\tDevice Name: {mnt.mnt_devname.string_().decode('utf-8')}")
+    if mnt.mnt_mp:
+        print(f"\tMount Point Count: {mnt.mnt_mp.m_count.value_()}")
+    else:
+        print("\tMount Point is NULL")
+    print(f"\tMount ID: {mnt.mnt_id}")
+    print(f"\tMount Group ID: {mnt.mnt_group_id}")
+    print(f"\tMount Expiry Mark: {mnt.mnt_expiry_mark}")
+    return
+
+def print_super_block_info(sb):
+    """Print key fields of `struct super_block`"""
+    print("Super Block:")
+    if not sb:
+        print("\tSuper block is NULL")
+        return
+    print(f"\tsb ref count: {sb.s_active.counter}")
+    print(f"\tsb s_count: {sb.s_count}")
+    print(f"\tsb umount rw_sem counter {sb.s_umount.count.counter}")
+    print(f"\tsb s_flags: {sb.s_flags}")
+    print(f"\tsb s_dev {sb.s_dev}")
+    print(f"\tsb s_id {sb.s_id}")
+    print(f"\tsb i_flags {sb.s_iflags}")
+
+    for mnt in list_for_each_entry("struct mount", sb.s_mounts.address_of_(), "mnt_instance"):
+        if not mnt.mnt_master:
+            print_mount_info(mnt)
+            break
+
+    return
+
+def print_fs_info(fs_info):
+    """Print key fields of `struct btrfs_fs_info`"""
+    print("FS Info:")
+    if not fs_info:
+        print("\tfs_info is NULL")
+        return
+
+    print(f"\tFS State: {fs_info.fs_state}")
+    print(f"\tFlags: {fs_info.flags}")
+    print(f"\tmount opt: {fs_info.mount_opt}")
+
+    print_super_block_info(fs_info.sb)
+    return
+
+def print_fs_devices_info(fs_dev):
+    """"Print key fields of `struct btrfs_fs_devices`"""
+    print("FS Devices:")
+    print(f"\tFS UUID: {get_uuid_from_list(list(fs_dev.fsid))}")
+    print(f"\tMETADATA UUID: {get_uuid_from_list(list(fs_dev.metadata_uuid))}")
+    print(f"\tfsid_change: {fs_dev.fsid_change}")
```

Comment:
torvalds/linux@5966930dfd7741cfa6a41cd182c7cf4dfc1a4c95 removed this field, so since Linux 6.7 this fails with:
```
AttributeError: 'struct btrfs_fs_devices' has no member 'fsid_change'
```

You can wrap it in a try/catch:

```suggestion
    try:
        print(f"\tfsid_change: {fs_dev.fsid_change}")
    except AttributeError:
        pass
```
Other than that, this looks reasonable, thanks!